### DISCUSSION TITLE
Thumb Combos not working after adding `custom_config.h` and `custom_rules.mk` files. QMK ferris/sweep

### DISCUSSION THREAD

**Discussion by dan90210** [manna-harbour/miryoku#256] [Q&A] [answered]
I'll preface with this is most likely an error on my part. When I run the following with no custom rules thumb combos seem to be enabled and I can access the escape key and media layer:

`make ferris/sweep:manna-harbour_miryoku:flash`

Below are my updates to the custom files. Essentially I just wanted to remap the thumb keys and enable vim navigation.

**_custom_config.h_**
```
#define MIRYOKU_LAYER_BASE \
KC_Q,              KC_W,              KC_F,              KC_P,              KC_B,              KC_J,              KC_L,              KC_U,              KC_Y,              KC_QUOT,           \
LGUI_T(KC_A),      LALT_T(KC_R),      LCTL_T(KC_S),      LSFT_T(KC_T),      KC_G,              KC_M,              LSFT_T(KC_N),      LCTL_T(KC_E),      LALT_T(KC_I),      LGUI_T(KC_O),      \
LT(U_BUTTON,KC_Z), ALGR_T(KC_X),      KC_C,              KC_D,              KC_V,              KC_K,              KC_H,              KC_COMM,           ALGR_T(KC_DOT),    LT(U_BUTTON,KC_SLSH),\
U_NP,              U_NP,              LT(U_MEDIA,KC_ESC),LT(U_NAV,KC_TAB),  LT(U_MOUSE,KC_ENT),LT(U_SYM,KC_SPC),  LT(U_NUM,KC_BSPC), LT(U_FUN,KC_DEL),  U_NP,              U_NP
```

**_custom_rules.mk_**
`MIRYOKU_NAV = VI`

After building with these files my keys are remapped and vim motions work, but it seems that the thumb combos are no longer enabled. I have tried adding `MIRYOKU_KLUDGE_THUMBCOMBOS=yes` to my `custom_rules.mk` file, but that didn't seem to fix the issue either. Any ideas where to start?

**Comment by davidsteinberger** [comment]
I'm  by no means an expert, but I think the problem is the rules.mk. Can you add `-e MIRYOKU_KLUDGE_THUMBCOMBOS=yes` to your make cmd?

**Reply by dan90210** [reply]
No dice. I ensured that my `custom_rules.mk` file only had the vi nav, then ran `make ferris/sweep:manna-harbour_miryoku:flash -e MIRYOKU_KLUDGE_THUMBCOMBOS=yes` and they still don't seem to be working. I'm curious if this is due to my remapping of the `MIRYOKU_LAYER_BASE`?. @manna-harbour do you have any thoughts?

**Comment by manna-harbour** [comment, marked as answer]
Ideally key emulation combos would be defined solely by position, but that isn't supported by QMK, so they are implemented based on keycodes instead. Thumb combos are only partially implemented and documented, and changing thumb combos in `custom_config.h` is not yet supported. See https://github.com/manna-harbour/miryoku/issues/56 and https://github.com/manna-harbour/miryoku/issues/94.

If you add or substitute custom layers you'll need to also manually modify the code to match by editing this section: https://github.com/manna-harbour/miryoku_qmk/blob/2b586786bbbf1a1dd23b2b178b36439cb64fcd51/users/manna-harbour_miryoku/manna-harbour_miryoku.c#L62.

**Reply by dan90210** [reply]
Thank you so much! I just needed to update the `base_right` and `base_left` combo definitions to match my current thumb layout and everything went off without a hitch.
